content/doc/book/pipeline/reference.adoc

```diff
@@ -0,0 +1,492 @@
+** Note that `[$class: 'Foo', arg1: 'something', ...]` syntax can not be
+used, only `cron('@daily')` and the like.
+** Also note that the `SCMTrigger` won't work with the `scm` `@Symbol` -
+with Jenkins 2.22 or later, the `pollScm` symbol does work.
+* _Examples_:
+
+[pipeline]
+----
+// Declarative //
+triggers {
+    cron('@daily')
+}
+----
+
+[[build-parameters]]
+=== Build Parameters
+
+* _Description_: Build parameters that will be prompted for at build
+time.
+* _Required_: No
+* _Allowed In_: Top-level `pipeline` closure only.
+* _Parameters_: None
+* _Takes a Closure_: Yes
+* _Closure Contents_: A sequence of one or more parameter definition
+configurations, using `@Symbol` names for constructors.
+** Note that `[$class: 'Foo', arg1: 'something', ...]` syntax can not be
+used, only `booleanParam(...)` and the like.
+* _Examples_:
+
+[pipeline]
+----
+// Declarative //
+parameters {
+    booleanParam(defaultValue: true, description: '', name: 'flag')
+    string(defaultValue: '', description: '', name: 'SOME_STRING')
+}
+----
+
+[[options]]
+=== Options
+
+* _Description_: Traditional `JobProperty`, such as `buildDiscarder` or
+`disableConcurrentBuilds`, Declarative-specifc options, such as
+`skipDefaultCheckout`, and "wrappers" that should wrap the entire build,
+such as `timeout`.
+* _Required_: No
+* _Allowed In_: Top-level `pipeline` closure only.
+* _Parameters_: None
+* _Takes a Closure_: Yes
+* _Closure Contents_: A sequence of one or more Declarative option or
+job property configurations, using `@Symbol` names for constructors.
+** Note that `[$class: 'Foo', arg1: 'something', ...]` syntax can not be
+used, only `booleanParam(...)` and the like.
+** Note that the `parameters` and `pipelineTriggers` `@Symbol` cannot
+be used here directly.
+* _Examples_:
+
+[pipeline]
+----
+// Declarative //
+options {
+    buildDiscarder(logRotator(numToKeepStr:'1'))
+    disableConcurrentBuilds()
+}
+----
+
+[[declarative-subset-of-groovy]]
+=== Declarative Subset of Groovy
+
+* Top-level has to be a block
+* No semicolons as statement separators. Each statement has to be on its
+own line
+* Block must only consists of method call statements, assignment
+statements, or property reference statement
+** A property reference statement is treated as no-arg method
+invocation. So for example, `input` is treated as `input()`
+* Expression has to be one of the following:
+** Literals (except class literals)
+** Numbers: `1`, `3`
+** Booleans: `true`, `false`
+** String literals regardless of their quotations: `"foo"`, `'bar'`
+** Multi-line string literals
+** Variable references: `x`
+** Sequence of property references: `x.y.z`
+** GString: `"hello ${exp}"`
+** Literal list: `[exp,exp,...]`
+** Literal map where keys are all constants: `[a:exp, b:exp, ... ]`
+** Method calls where the left hand side is a variable reference or a
+sequence of property references: `x.y.z(...)`
+** Method calls (including `@Symbol` constructors like used above in
```

Comment:
Lines 489 to 491-- missing word, and extraneous word. Better: "...constructors like those used above in options, triggers, and build parameters) where there is no left side"

Direction, whether left or right, does not need to be anchored in handedness.